pfcpiface/pfcpiface.go

```diff
@@ -358,14 +358,40 @@ func handleSessionEstablishmentRequest(upf *upf, msg message.Message, addr net.A
 	ctx, cancel := context.WithTimeout(context.Background(), Timeout)
 	defer cancel()
 	done := make(chan bool)
-	upf.pauseAll()
+	calls := len(pdrs) + len(fars)
+	boom := time.After(Timeout)
+	err = upf.pauseAll()
+	if err != nil {
+		log.Fatalln("Unable to pause BESS:", err)
+	}
 	for _, pdr := range pdrs {
 		upf.addPDR(ctx, done, pdr)
 	}
 	for _, far := range fars {
 		upf.addFAR(ctx, done, far)
 	}
-	upf.resumeAll()
+	for {
+		select {
+		case ok := <-done:
+			if !ok {
+				log.Println("Error adding PDRs and/or FARs")
+				cancel()
+				break
+			}
+			calls = calls - 1
+		case <-boom:
+			log.Println("Timed out adding PDRs and/or FARs")
+			break
+		}
+		if calls == 0 {
+			break
+		}
+	}
+
+	err = upf.resumeAll()
+	if err != nil {
+		log.Fatalln("Unable to resume BESS:", err)
+	}
```

Comment:
we should make a function within upf.go common for sim as well as pfcpiface to leverage, since this feels like internal details of bess leaking into pfcpiface.go